server/plugin.go

```diff
@@ -18,11 +36,155 @@ type Plugin struct {
 	// configuration is the active plugin configuration. Consult getConfiguration and
 	// setConfiguration for usage.
 	configuration *configuration
+
+	postsWaitingForOnline map[string][]*model.Post
+	deferedPosts          []*DeferedPost
+	Queues                map[string]*Queue
 }
 
 // ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
 func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Request) {
-	fmt.Fprint(w, "Hello, world!")
+	userID := r.Header.Get("Mattermost-User-ID")
+	fmt.Println(p.postsWaitingForOnline[userID])
+	if posts, ok := p.postsWaitingForOnline[userID]; ok && posts != nil {
+		for _, post := range posts {
+			p.API.CreatePost(post)
+		}
+		p.postsWaitingForOnline[userID] = nil
+	}
+	fmt.Fprint(w, "{}")
 }
 
-// See https://developers.mattermost.com/extend/plugins/server/reference/
+func (p *Plugin) OnActivate() error {
+	err := p.RestoreWaitingForOnlinePosts()
+	if err != nil {
+		p.API.LogError(err.Error())
```

Comment:
Should these logs have more context? Something like:

```suggestion
		p.API.LogError("failed to restore \"waiting for online\" posts", "err", err.Error())
```

Same with other cases that we just log `p.API.LogError(err.Error())`